listenbrainz/db/user_timeline_event.py

```diff
@@ -168,3 +190,62 @@ def get_user_notification_events(user_id: int, count: int = 50) -> List[UserTime
         event_type=UserTimelineEventType.NOTIFICATION,
         count=count
     )
+
+
+def hide_user_timeline_event(user_id: int, event_type: UserTimelineEventType, event_id: int) -> bool:
+    """ Adds events that are to be hidden """
+    try:
+        with db.engine.connect() as connection:
+            result = connection.execute(sqlalchemy.text('''
+                INSERT INTO hide_user_timeline_event (user_id, event_type, event_id)
+                    VALUES (:user_id, :event_type, :event_id)
+                ON CONFLICT (user_id, event_type, event_id)
+                DO NOTHING
+            '''), {
+                'user_id': user_id,
+                'event_type': event_type,
+                'event_id': event_id
+                }
+            )
+            return result.rowcount == 1
+    except Exception as e:
+        raise DatabaseException(str(e))
+
+
+def get_hidden_timeline_events(user: int, count: int) -> List[HiddenUserTimelineEvent]:
+    '''Retrieves all events that are hidden by the user, based on event_type'''
+    try:
+        with db.engine.connect() as connection:
+            result = connection.execute(sqlalchemy.text('''
+                SELECT * FROM hide_user_timeline_event WHERE
+                user_id = :user_id
+                ORDER BY created DESC
+                LIMIT :count
+            '''), {
+                'user_id': user,
+                'count': count
+                }
+            )
+            return [HiddenUserTimelineEvent(**row) for row in result.fetchall()]
+    except Exception as e:
+        raise DatabaseException(str(e))
+
+
+def unhide_timeline_event(user: int, event_type: UserTimelineEventType, event_id: int) -> bool:
+    ''' Deletes hidden timeline events for a user with specific row id '''
+    try:
+        with db.engine.connect() as connection:
+            result = connection.execute(sqlalchemy.text('''
+                DELETE FROM hide_user_timeline_event WHERE
+                user_id = :user_id AND
+                event_type = :event_type AND
+                event_id = :event_id
+            '''), {
+                'user_id': user,
+                'event_type': event_type,
+                'event_id': event_id
+                }
+            )
+            return result.rowcount == 1
```

Comment:
We don't verify in the API that a user hasn't already hidden an event so same event can have multiple entries in table. So better to rewrite as this > 0.